.NET/Microsoft.Recognizers.Text.Choice/Arabic/Extractors/ArabicBooleanExtractorConfiguration.cs

```diff
@@ -43,5 +45,7 @@ public ArabicBooleanExtractorConfiguration(bool onlyTopMatch = true)
         public int MaxDistance => 2;
 
         public bool OnlyTopMatch { get; }
+
+        protected static TimeSpan RegexTimeOut => ChoiceRecognizer.GetTimeout(MethodBase.GetCurrentMethod().DeclaringType);
```

Comment:
Would introducing an abstract class allow you to remove this line from all the classes? So, instead of IBooleanExtractorConfiguration you have AbstractBooleanExtractorConfiguration which implements the interface and provides the RegexTimeout.

Reply:
Nice suggestion. The refactoring done.